### PR TITLE
chore(refactor): move cspc code v1alpha1 package

### PR DESCRIPTION
**Pull request by shovanmaity**
This PR -

- Moves CSPC builder code to v1alpha1 package
- Moves localdevice controller and finalizer to v1alpha1 package
- Moves metac.config to v1alpha1 directory

Here is the part of sts using this you can create CSPC of v1alpha1
using cstorpoolauto
```yaml
containers:
- args
  - --logtostderr
  - --run-as-local
  - -v=5
  - --discovery-interval=40s
  - --cache-flush-interval=240s
  - --metac-config-path=/etc/config/metac/localdevice/v1alpha1/
  command:
  - /usr/bin/cstorpoolauto
  image: mayadataio/cstorpoolauto:<>
  name: cstorpoolauto
```

### CODE REVIEW

**Comment by AmitKumarDas** [comment]
@shovanmaity is this required?
`Move metac.config to v1alpha1 directory`

**Comment by shovanmaity** [comment]
> @shovanmaity is this required?
> `Move metac.config to v1alpha1 directory`
I think yes if we want to create both `v1` and `v1alpha1` CSPC with a single image.
I mean we need to point different controller and finalizer based on version.
cc @AmitKumarDas

**Comment by shovanmaity** [comment]
So the user-facing changes will be this
If you want to create v1alpha1 cspc then use
```yaml
containers:
- args
  - --logtostderr
  - --run-as-local
  - -v=5
  - --discovery-interval=40s
  - --cache-flush-interval=240s
  - --metac-config-path=/etc/config/metac/localdevice/v1alpha1/
  command:
  - /usr/bin/cstorpoolauto
  image: mayadataio/cstorpoolauto:<>
  name: cstorpoolauto
```

If you want to create v1 cspc then use
```yaml
containers:
- args
  - --logtostderr
  - --run-as-local
  - -v=5
  - --discovery-interval=40s
  - --cache-flush-interval=240s
  - --metac-config-path=/etc/config/metac/localdevice/
  command:
  - /usr/bin/cstorpoolauto
  image: mayadataio/cstorpoolauto:<>
  name: cstorpoolauto
```

**Comment by AmitKumarDas** [comment]
Will there be cases to support both versions?
If that happens, how to handle such.
IMO lets handle that now itself.
Probably by giving a different GenericController name in the config

**Comment by AmitKumarDas** [comment]
OTH we need to get back to the concept of a single config file.
Once we have tagging support etc in metac, we need to converge to a single file.
Presence of single config file should not ensure all the GenericControllers are loaded. They should be based on tags etc.

File: cmd/main.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 	"mayadata.io/cstorpoolauto/controller/cstorclusterplan"
 	"mayadata.io/cstorpoolauto/controller/cstorclusterstorageset"
 	"mayadata.io/cstorpoolauto/controller/cstorpoolcluster"
-	"mayadata.io/cstorpoolauto/controller/localdevice"
+	localdevice "mayadata.io/cstorpoolauto/controller/localdevice/v1alpha1"
```
Comment (AmitKumarDas) [inline cmd/main.go:28]: Does `v1alpha1-localdevice` look better?

Comment (AmitKumarDas) [reply cmd/main.go:28]: or perhaps `localdevice-v1alpha1`

File: common/cstorpoolcluster/v1alpha1/helper_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cstorpoolcluster
+package v1alpha1
```
Comment (AmitKumarDas) [inline common/cstorpoolcluster/v1alpha1/helper_test.go:17]: do we really need to change the package name? lets avoid

File: common/cstorpoolcluster/v1alpha1/builder.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cstorpoolcluster
+package v1alpha1
```
Comment (AmitKumarDas) [inline common/cstorpoolcluster/v1alpha1/builder.go:17]: Do we really need to change the package name? lets avoid.

File: common/cstorpoolcluster/v1alpha1/builder_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cstorpoolcluster
+package v1alpha1
```
Comment (AmitKumarDas) [inline common/cstorpoolcluster/v1alpha1/builder_test.go:17]: Do we really need to change the package name? lets avoid.

File: common/cstorpoolcluster/v1alpha1/helper.go [hunk 5]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cstorpoolcluster
+package v1alpha1
```
Comment (AmitKumarDas) [inline common/cstorpoolcluster/v1alpha1/helper.go:17]: Do we really need to change the package name? lets avoid.

File: controller/localdevice/v1alpha1/finalizer.go [hunk 6]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package localdevice
+package v1alpha1
```
Comment (AmitKumarDas) [inline controller/localdevice/v1alpha1/finalizer.go:17]: Do we really need to change the package name? lets avoid.

File: controller/localdevice/v1alpha1/finalizer_test.go [hunk 7]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package localdevice
+package v1alpha1
```
Comment (AmitKumarDas) [inline controller/localdevice/v1alpha1/finalizer_test.go:17]: Do we really need to change the package name? lets avoid.

File: controller/localdevice/v1alpha1/reconciler.go [hunk 8]
```diff
@@ -27,7 +27,7 @@ import (
 
 	bd "mayadata.io/cstorpoolauto/common/blockdevice"
 	ccc "mayadata.io/cstorpoolauto/common/cstorclusterconfig"
-	cspc "mayadata.io/cstorpoolauto/common/cstorpoolcluster"
+	cspc "mayadata.io/cstorpoolauto/common/cstorpoolcluster/v1alpha1"
```
Comment (AmitKumarDas) [inline controller/localdevice/v1alpha1/reconciler.go:30]: How about `cspc-v1alpha1`?

File: controller/localdevice/v1alpha1/reconciler.go [hunk 9]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package localdevice
+package v1alpha1
```
Comment (AmitKumarDas) [inline controller/localdevice/v1alpha1/reconciler.go:17]: Lets avoid renaming the package

File: controller/localdevice/v1alpha1/reconciler.go (hunk 8 above)
Comment (shovanmaity) [reply controller/localdevice/v1alpha1/reconciler.go:30]: As this reconciler is a part of `v1alpha1` package I think we don't need this.

**Comment by shovanmaity** [comment]
> Will there be cases to support both versions?
If that happens, how to handle such.
IMO lets handle that now itself.
Probably by giving a different GenericController name in the config

I thought for that(both version) only we made `v1alpha1` package else we can directly make changes in existing code.

**Comment by AmitKumarDas** [comment]
> > Will there be cases to support both versions?
> > If that happens, how to handle such.
> > IMO lets handle that now itself.
> > Probably by giving a different GenericController name in the config
> 
> I thought for that(both version) only we made `v1alpha1` package else we can directly make changes in existing code.

It is to support both versions. However, there has to be one config to support both from a single controller image.

File: config/localdevice/metac.yaml [hunk 10]
```diff
@@ -1,60 +0,0 @@
-apiVersion: metac.openebs.io/v1alpha1
```
Comment (AmitKumarDas) [inline config/localdevice/metac.yaml:1]: I wanted us to use this file only. With 2 controllers. One controller for v1alpha1 & other for the upcoming v1.

File: config/localdevice/v1alpha1/metac.yaml [hunk 11]
```diff
@@ -0,0 +1,59 @@
+apiVersion: metac.openebs.io/v1alpha1
+kind: GenericController
+metadata:
+  name: sync-localdevice
```
Comment (AmitKumarDas) [inline config/localdevice/v1alpha1/metac.yaml:4]: can we rename this to sync-localdevice-v1alpha1

File: config/localdevice/v1alpha1/metac.yaml [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    # controller gets triggered through this hook when
+    # CStorClusterConfig gets created or modified
+    sync:
+      inline:
+        funcName: sync/localdevicev1alpha1
+---
+apiVersion: metac.openebs.io/v1alpha1
+kind: GenericController
+metadata:
+  name: finalize-localdevice
```
Comment (AmitKumarDas) [inline config/localdevice/v1alpha1/metac.yaml:36]: can we rename this to finalize-localdevice-v1alpha1

**Pull request merged**